config/base/dex/numaflow-dex-server-deployment.yaml

```diff
@@ -16,6 +16,38 @@ spec:
         app.kubernetes.io/component: dex-server
     spec:
       serviceAccountName: numaflow-dex-server
+      initContainers:
+      - name: dex-init
+        image: quay.io/numaproj/numaflow:latest
+        imagePullPolicy: Always
+        args:
+        - dex-server-init
+        volumeMounts:
+        - mountPath: /cfg
+          name: connector-config
+        - mountPath: /tls
+          name: tls
+        - mountPath: /tmp
+          name: generated-dex-config
+        env:
+        - name: NUMAFLOW_SERVER_DEX_TLS
```

Comment:
Can we clean up this throughly? we could keep the source code to support disabling tls, but don't want to open the entry in the config for now.